src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -262,30 +262,40 @@ const connectInfiniteHits: InfiniteHitsConnector = (
       },
 
       getWidgetState(uiState, { searchParameters }) {
-        const page = searchParameters.page || 0;
-
-        if (!hasShowPrevious || page === 0 || page + 1 === uiState.page) {
+        if (!hasShowPrevious) {
           return uiState;
         }
 
+        // The page in the UI state is incremented by one
+        // to expose the user value (not `0`).
+        const page = (searchParameters.page || 0) + 1;
+
         return {
           ...uiState,
-          page: page + 1,
+          page,
         };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        if (!hasShowPrevious) {
-          return searchParameters;
+        let state = searchParameters;
```

Comment:
I would have kept the same pattern used in `dispose` with named variable vs `let`.